ledger-service/cli-opts/src/main/scala/cliopts/Metrics.scala

```diff
@@ -6,9 +6,28 @@ package com.daml.cliopts
 import com.daml.metrics.MetricsReporter
 import scopt.OptionDef
 
-import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration.{Duration, FiniteDuration, NANOSECONDS}
+import scala.util.Try
 
 object Metrics {
+  private case class DurationFormat(unwrap: FiniteDuration)
+
+  private implicit val scoptDurationFormat: scopt.Read[DurationFormat] = scopt.Read.reads {
+    duration =>
+      Try {
+        Duration(duration)
+      }.orElse(Try {
+        Duration.fromNanos(
+          java.time.Duration.parse(duration).toNanos
```

Comment:
Maybe add a comment that this is for backwards compat. I’d also be slightly more comfortable trying the java version first. I think there is no collision where something parses as both but if there is we want to prefer Java.